### ISSUE TITLE
[BUG] Docker Image Paperless-ngx ARM7: Django doesn't find any language

### ISSUE THREAD

**Issue by datensenke** [paperless-ngx/paperless-ngx#4486]
### Description

Under Archlinux ARM, when I do a docker-compose up -d, I get the error message:
`?: The selected ocr language eng is not installed. Paperless cannot OCR your documents without it. Please fix PAPERLESS_OCR_LANGUAGE.`

[docker-compose.env.txt](https://github.com/paperless-ngx/paperless-ngx/files/13238135/docker-compose.env.txt)
[docker-compose.yml.txt](https://github.com/paperless-ngx/paperless-ngx/files/13238146/docker-compose.yml.txt)

The directory
`drwxr-xr-x  4 alarm alarm  120  2. Nov 12:14 .`
`drwxrwxrwx 20 alarm alarm 4096  1. Nov 15:36 ..`
`-rw-r--r--  1 alarm alarm 2065  1. Nov 20:48 2`
`drwxrwxr-x  2 alarm alarm    6  2. Nov 11:54 consume`
`-rw-r--r--  1 alarm alarm 2065  1. Nov 21:54 docker-compose.env`
`-rw-r--r--  1 alarm alarm 2921  1. Nov 21:52 docker-compose.yml`
`drwxrwxr-x  2 alarm alarm    6  1. Nov 16:11 export`

User and group alarm is 1000.
The same setup has no problems under Archlinux x86.

### Steps to reproduce

docker-compose up -d

### Webserver logs

```bash
[+] Building 0.0s (0/0)                                                                                                                                                                                                                      docker:default
[+] Creating 2/0
 ✔ Container paperless-gotenberg-1  Running                                                                                                                                                                                                            0.0s 
 ✔ Container paperless-broker-1     Running                                                                                                                                                                                                            0.0s 
[+] Running 1/0
 ✔ Container paperless-tika-1  Started                                                                                                                                                                                                                 0.0s 
[+] Building 0.0s (0/0)                                                                                                                                                                                                                      docker:default
Paperless-ngx docker container starting...
Installing languages...
Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
Get:2 http://deb.debian.org/debian bookworm-updates InRelease [52.1 kB]
Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
Get:4 http://deb.debian.org/debian bookworm/main armhf Packages [8493 kB]
Get:5 http://deb.debian.org/debian bookworm-updates/main armhf Packages [6388 B]
Get:6 http://deb.debian.org/debian-security bookworm-security/main armhf Packages [87.7 kB]
Fetched 8838 kB in 4s (2502 kB/s)                          
Reading package lists... Done
Package tesseract-ocr-deu already installed!
Package tesseract-ocr-eng already installed!
Creating directory /tmp/paperless
Adjusting permissions of paperless files. This may take a while.
Waiting for Redis...
Connected to Redis broker.
Apply database migrations...
Operations to perform:
  Apply all migrations: admin, auth, authtoken, contenttypes, django_celery_results, documents, guardian, paperless_mail, sessions
Running migrations:
  No migrations to apply.
Running Django checks
SystemCheckError: System check identified some issues:

ERRORS:
?: The selected ocr language eng is not installed. Paperless cannot OCR your documents without it. Please fix PAPERLESS_OCR_LANGUAGE.

System check identified 1 issue (0 silenced).
```


### Browser logs

_No response_

### Paperless-ngx version

1.7.4, 1.7.2, 1.6

### Host OS

Archlinux ARM

### Installation method

Docker - official image

### Browser

_No response_

### Configuration changes

_No response_

### Other

_No response_

**Comment by stumpylog** [comment]
The code is the exact same, so I would look at your configuration.  Make sure the formatting is correct, using the correct separators.  Your provided environment for doesn't actually set eng as an ocr language anyway, so that's already a difference.

**Comment by stumpylog** [comment]
The image is the same image, based on Debian.  You can run `tesseract --list-langs` yourself in the container, but given no one else has encountered this, I remain inclined to think it is something with your particular system.

**Comment by datensenke** [comment]
I tried it again and now it works. Yesterday I had all the time the same issue. This is strange. I have to investigate it further, but you are right: it is not your issue.

**Comment by datensenke** [comment]
Sorry to write that, but I was too optimistic. The docker-compose didn't write the error message, because this time I started it with `-d`.
Now, I had a look into the logs `docker logs -f ...` and I still get the same error message.

**Comment by datensenke** [comment]
`docker-compose run --rm webserver createsuperuser`
returns the output:
[+] Building 0.0s (0/0)                                                                                                                                                                                                                      docker:default
[+] Creating 2/0
 ✔ Container paperless-broker-1     Running                                                                                                                                                                                                            0.0s 
 ✔ Container paperless-gotenberg-1  Running                                                                                                                                                                                                            0.0s 
[+] Running 1/0
 ✔ Container paperless-tika-1  Started                                                                                                                                                                                                                 0.0s 
[+] Building 0.0s (0/0)                                                                                                                                                                                                                      docker:default
Paperless-ngx docker container starting...
Installing languages...
Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
Get:2 http://deb.debian.org/debian bookworm-updates InRelease [52.1 kB]
Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
Get:4 http://deb.debian.org/debian bookworm/main armhf Packages [8493 kB]
Get:5 http://deb.debian.org/debian bookworm-updates/main armhf Packages [6388 B]
Get:6 http://deb.debian.org/debian-security bookworm-security/main armhf Packages [87.7 kB]
Fetched 8838 kB in 3s (2608 kB/s)                        
Reading package lists... Done
Package tesseract-ocr-deu already installed!
Package tesseract-ocr-eng already installed!
Creating directory /tmp/paperless
Adjusting permissions of paperless files. This may take a while.
Waiting for Redis...
Connected to Redis broker.
Apply database migrations...
Operations to perform:
  Apply all migrations: admin, auth, authtoken, contenttypes, django_celery_results, documents, guardian, paperless_mail, sessions
Running migrations:
  No migrations to apply.
Running Django checks
SystemCheckError: System check identified some issues:

ERRORS:
?: The selected ocr language deu is not installed. Paperless cannot OCR your documents without it. Please fix PAPERLESS_OCR_LANGUAGE.

**Comment by datensenke** [comment]
> The image is the same image, based on Debian. You can run `tesseract --list-langs` yourself in the container, but given no one else has encountered this, I remain inclined to think it is something with your particular system.

How can I run `tesseract --list-langs` if it won't start the container?
Maybe, I'm the only trying to run it on ARM7?

**Comment by sippeangelo** [comment]
I'm experiencing this too just now when I tried to migrate from paperless-ng. Before eng was assumed to be installed by default, but now it wouldn't launch so I added it to `PAPERLESS_OCR_LANGUAGES` to force-install it:
```
PAPERLESS_OCR_LANGUAGES=swe eng
PAPERLESS_OCR_LANGUAGE=swe+eng
```

Now the log is literally saying that eng is already installed, but it then it also still says that it's missing:

> paperless-webserver-1  | Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
> paperless-webserver-1  | Get:2 http://deb.debian.org/debian bookworm-updates InRelease [52.1 kB]
> paperless-webserver-1  | Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
> paperless-webserver-1  | Get:4 http://deb.debian.org/debian bookworm/main amd64 Packages [8780 kB]
> paperless-webserver-1  | Get:5 http://deb.debian.org/debian bookworm-updates/main amd64 Packages [6408 B]
> paperless-webserver-1  | Get:6 http://deb.debian.org/debian-security bookworm-security/main amd64 Packages [92.8 kB]
> paperless-webserver-1  | Fetched 9131 kB in 1s (10.1 MB/s)
> paperless-webserver-1  | Reading package lists...
> paperless-webserver-1  | Installing package tesseract-ocr-swe...
> paperless-webserver-1  | **Package tesseract-ocr-eng already installed!**
> paperless-webserver-1  | Creating directory /tmp/paperless
> paperless-webserver-1  | Adjusting permissions of paperless files. This may take a while.
> paperless-webserver-1  | Waiting for Redis...
> paperless-webserver-1  | Connected to Redis broker.
> paperless-webserver-1  | Apply database migrations...
> paperless-webserver-1  | Operations to perform:
> paperless-webserver-1  |   Apply all migrations: admin, auth, authtoken, contenttypes, django_celery_results, documents, guardian, paperless_mail, sessions
> paperless-webserver-1  | Running migrations:
> paperless-webserver-1  |   No migrations to apply.
> paperless-webserver-1  | Running Django checks
> paperless-webserver-1  | SystemCheckError: System check identified some issues:
> paperless-webserver-1  | 
> paperless-webserver-1  | ERRORS:
> paperless-webserver-1  | ?: **The selected ocr language eng is not installed. Paperless cannot OCR your documents without it. Please fix PAPERLESS_OCR_LANGUAGE.**
> paperless-webserver-1  | 
> paperless-webserver-1  | System check identified 1 issue (0 silenced).

**Comment by stumpylog** [comment]
You'll have to override the entrypoint, `docker run --rm -it --entrypoint /bin/bash ghcr.io/paperless-ngx/paperless-ngx:dev` for example

**Comment by datensenke** [comment]
`ls /usr/share/tesseract-ocr/5/tessdata/`
`configs  deu.traineddata  eng.traineddata  fra.traineddata  ita.traineddata  osd.traineddata  pdf.ttf  spa.traineddata	tessconfigs`

'tesseract --list-langs`
`List of available languages in "/usr/share/tesseract-ocr/5/tessdata/" (0):`

**Comment by stumpylog** [comment]
What does `dpkg --list | grep tesseract-ocr` return?

(also please use backticks to format, it's very hard to tell what the commands and output run are)

**Comment by datensenke** [comment]
how can I run the python function I mentioned before?

**Comment by sippeangelo** [comment]
I believe in my case this is due to my old paperless-ng docker setup having a volume set up for tesseract so it won't redownload each time, and it previously having using version 4 of tesseract? Once I nuke the volume it might be ok. Might be worth to mention in the migration guide from paperless-ng unless it happens normally too.

**Comment by stumpylog** [comment]
Three backticks, a new line, then the code, a new line and three back ticks.  I edited your comment to use that.

This appears as a tesseract problem.  The system has the language packages installed, at the expected versions and location.  But tesseract itself isn't seeing them.  We rely on tesseract to find them, so there's not more we can do from here.

> having a volume set up for tesseract

That is a very specific case that I don't think is common enough to worth mentioning.  yes, you should adjust/re-create that for Tesseract 5

**Comment by datensenke** [comment]
I understand. I thought, the message `Package tesseract-ocr-deu already installed!` is from tesseract, but it is from dpkg.

**Comment by stumpylog** [comment]
What happens if you force re-install of the package, then list languages?  Essentially `apt-get update && apt-get --reinstall tesseract-ocr-eng && tesseract --list-langs`

**Comment by datensenke** [comment]
> What happens if you force re-install of the package, then list languages? Essentially `apt-get update && apt-get --reinstall tesseract-ocr-eng && tesseract --list-langs`

Same.

**Comment by datensenke** [comment]
If I create a directory `/usr/share/tessdata/` and I copy everything from `/usr/share/tesseract-ocr/5/tessdata` to `/usr/share/tessdata/` and set 'export TESSDATA_PREFIX=/usr/share/tessdata/`, then `tesseract --list-langs` works.

**Comment by stumpylog** [comment]
What if you set `TESSDATA_PREFIX=/usr/share/tesseract-ocr/5/tessdata`?

**Comment by datensenke** [comment]
> What if you set `TESSDATA_PREFIX=/usr/share/tesseract-ocr/5/tessdata`?

Same error as before. I even tried `chmod aug+rwX`. Nothing.

**Comment by datensenke** [comment]
```
root@60fa173e1c5d:~# export
declare -x GPG_KEY="E3FF2839C048B25C084DEBE9B26995E310250568"
declare -x HOME="/root"
declare -x HOSTNAME="60fa173e1c5d"
declare -x LANG="C.UTF-8"
declare -x OLDPWD="/usr/share/tesseract-ocr/5/tessdata"
declare -x PATH="/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
declare -x PWD="/root"
declare -x PYTHON_GET_PIP_SHA256="45a2bb8bf2bb5eff16fdd00faef6f29731831c7c59bd9fc2bf1f3bed511ff1fe"
declare -x PYTHON_GET_PIP_URL="https://github.com/pypa/get-pip/raw/9af82b715db434abb94a0a6f3569f43e72157346/public/get-pip.py"
declare -x PYTHON_PIP_VERSION="23.0.1"
declare -x PYTHON_SETUPTOOLS_VERSION="58.1.0"
declare -x PYTHON_VERSION="3.9.18"
declare -x SHLVL="1"
declare -x TERM="xterm"
declare -x TESSDATA_PREFIX=""
root@60fa173e1c5d:~# unset TESSDATA_PREFIX
root@60fa173e1c5d:~# tesseract --list-langs
List of available languages in "/usr/share/tesseract-ocr/5/tessdata/" (0):
root@60fa173e1c5d:~# export
declare -x GPG_KEY="E3FF2839C048B25C084DEBE9B26995E310250568"
declare -x HOME="/root"
declare -x HOSTNAME="60fa173e1c5d"
declare -x LANG="C.UTF-8"
declare -x OLDPWD="/usr/share/tesseract-ocr/5/tessdata"
declare -x PATH="/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
declare -x PWD="/root"
declare -x PYTHON_GET_PIP_SHA256="45a2bb8bf2bb5eff16fdd00faef6f29731831c7c59bd9fc2bf1f3bed511ff1fe"
declare -x PYTHON_GET_PIP_URL="https://github.com/pypa/get-pip/raw/9af82b715db434abb94a0a6f3569f43e72157346/public/get-pip.py"
declare -x PYTHON_PIP_VERSION="23.0.1"
declare -x PYTHON_SETUPTOOLS_VERSION="58.1.0"
declare -x PYTHON_VERSION="3.9.18"
declare -x SHLVL="1"
declare -x TERM="xterm"
root@60fa173e1c5d:~# tesseract --list-langs
List of available languages in "/usr/share/tesseract-ocr/5/tessdata/" (0):
root@60fa173e1c5d:~# export TESSDATA_PREFIX=/usr/share/tessdata/
root@60fa173e1c5d:~# tesseract --list-langs
List of available languages in "/usr/share/tessdata/" (6):
deu
eng
fra
ita
osd
spa
root@60fa173e1c5d:~# export
declare -x GPG_KEY="E3FF2839C048B25C084DEBE9B26995E310250568"
declare -x HOME="/root"
declare -x HOSTNAME="60fa173e1c5d"
declare -x LANG="C.UTF-8"
declare -x OLDPWD="/usr/share/tesseract-ocr/5/tessdata"
declare -x PATH="/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
declare -x PWD="/root"
declare -x PYTHON_GET_PIP_SHA256="45a2bb8bf2bb5eff16fdd00faef6f29731831c7c59bd9fc2bf1f3bed511ff1fe"
declare -x PYTHON_GET_PIP_URL="https://github.com/pypa/get-pip/raw/9af82b715db434abb94a0a6f3569f43e72157346/public/get-pip.py"
declare -x PYTHON_PIP_VERSION="23.0.1"
declare -x PYTHON_SETUPTOOLS_VERSION="58.1.0"
declare -x PYTHON_VERSION="3.9.18"
declare -x SHLVL="1"
declare -x TERM="xterm"
declare -x TESSDATA_PREFIX="/usr/share/tessdata/"

```

**Comment by stumpylog** [comment]
Very strange.  The only thing I can imagine is that Tesseract for an arm device is configured differently in where it looks.  The only solution would be a [custom startup script](https://docs.paperless-ngx.com/advanced_usage/#custom-container-initialization) to copy the files or link the directories via symlink

**Comment by datensenke** [comment]
I tried it again, with a new container and my method didn't work. That is very strange. The custom script wasn't started anyway, so this would not have helped, too.
I tried out tesseract on my ARM machine and everything works fine. So, this is not a tesseract issue. Maybe my docker-compose setup is bad.

**Comment by datensenke** [comment]
Good news! Since these errors looked like random behaviour, I pruned the docker system and reinstalled the stuff from docker-compose again. Now it works!

But, tika is complaining about not having the right JVM.
``Error: missing `server' JVM at `/usr/lib/jvm/java-17-openjdk-armhf/lib/server/libjvm.so'. I'll have a look into that. 

I think the issue I wrote can be closed. The error was somewhere between the eyes and the back of the head.
